ci/unit/docker-setup.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+# This is intended to be run the plugin's root directory. `ci/unit/docker-test.sh`
+# Ensure you have Docker installed locally and set the ELASTIC_STACK_VERSION environment variable.
+set -e
+
+VERSION_URL="https://raw.githubusercontent.com/elastic/logstash/master/ci/logstash_releases.json"
+
+if [ "$ELASTIC_STACK_VERSION" ]; then
+    echo "Fetching versions from $VERSION_URL"
+    VERSIONS=$(curl $VERSION_URL)
+    if [[ "$SNAPSHOT" = "true" ]]; then
+      ELASTIC_STACK_RETRIEVED_VERSION=$(echo $VERSIONS | jq '.snapshots."'"$ELASTIC_STACK_VERSION"'"')
+      echo $ELASTIC_STACK_RETRIEVED_VERSION
+    else
+      ELASTIC_STACK_RETRIEVED_VERSION=$(echo $VERSIONS | jq '.releases."'"$ELASTIC_STACK_VERSION"'"')
+    fi
+    if [[ "$ELASTIC_STACK_RETRIEVED_VERSION" != "null" ]]; then
+      # remove starting and trailing double quotes
+      ELASTIC_STACK_RETRIEVED_VERSION="${ELASTIC_STACK_RETRIEVED_VERSION%\"}"
+      ELASTIC_STACK_RETRIEVED_VERSION="${ELASTIC_STACK_RETRIEVED_VERSION#\"}"
+      echo "Translated $ELASTIC_STACK_VERSION to ${ELASTIC_STACK_RETRIEVED_VERSION}"
+      export ELASTIC_STACK_VERSION=$ELASTIC_STACK_RETRIEVED_VERSION
+    fi
+
+    echo "Testing against version: $ELASTIC_STACK_VERSION"
+
+    if [[ "$ELASTIC_STACK_VERSION" = *"-SNAPSHOT" ]]; then
+        cd /tmp
+        wget https://snapshots.elastic.co/docker/logstash-"$ELASTIC_STACK_VERSION".tar.gz
+        tar xfvz logstash-"$ELASTIC_STACK_VERSION".tar.gz  repositories
+        echo "Loading docker image: "
+        cat repositories
+        docker load < logstash-"$ELASTIC_STACK_VERSION".tar.gz
+        rm logstash-"$ELASTIC_STACK_VERSION".tar.gz
+        cd -
+    fi
+
+    if [ -f Gemfile.lock ]; then
+        rm Gemfile.lock
+    fi
+
+    docker-compose -f ci/unit/docker-compose.yml down
+    docker-compose -f ci/unit/docker-compose.yml build
+    #docker-compose -f ci/unit/docker-compose.yml up --exit-code-from logstash --force-recreate
+else
```

Comment:
nitpick: when the `else` clause serves to emit an error/exit based on the condition applied to the `if` clause, I find that inverting the condition makes the code more readable (it eliminates need for nesting, and keeps error messages close to what caused them).

~~~ sh
if [ -z "${ELASTIC_STACK_VERSION}" ]; then
    echo "Please set the ELASTIC_STACK_VERSION environment variable"
    echo "For example: export ELASTIC_STACK_VERSION=6.2.4"
    exit 1
fi
~~~